src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -38,162 +34,102 @@ NonClientIslandWindow::~NonClientIslandWindow()
 {
 }
 
-// Method Description:
-// - Used to initialize the XAML island for the non-client area. Also calls our
-//   base IslandWindow's Initialize, which will initialize the client XAML
-//   Island.
-void NonClientIslandWindow::Initialize()
+void NonClientIslandWindow::OnDragBarSizeChanged(winrt::Windows::Foundation::IInspectable sender, winrt::Windows::UI::Xaml::SizeChangedEventArgs eventArgs)
 {
-    _nonClientSource = DesktopWindowXamlSource{};
-    auto interop = _nonClientSource.as<IDesktopWindowXamlSourceNative>();
-    winrt::check_hresult(interop->AttachToWindow(_window));
-
-    // stash the child interop handle so we can resize it when the main hwnd is resized
-    interop->get_WindowHandle(&_nonClientInteropWindowHandle);
-
-    _nonClientRootGrid = winrt::Windows::UI::Xaml::Controls::Grid{};
-
-    _nonClientSource.Content(_nonClientRootGrid);
-
-    // Call the IslandWindow Initialize to set up the client xaml island
-    IslandWindow::Initialize();
+    InvalidateRect(NULL, NULL, TRUE);
+    ForceResize();
 }
 
-// Method Description:
-// - Sets the content of the non-client area of our window to the given XAML element.
-// Arguments:
-// - content: a XAML element to use as the content of the titlebar.
-// Return Value:
-// - <none>
-void NonClientIslandWindow::SetNonClientContent(winrt::Windows::UI::Xaml::UIElement content)
+void NonClientIslandWindow::OnAppInitialized(winrt::TerminalApp::App app)

```

Comment:
Could we leave this method as taking a `UIElement`, and somehow have the `AppHost` _tell_ the `NonClientIslandWindow` about the dragbar? IIRC there aren't any other references to the `TerminalApp::App` in the `*IslandWindow` classes

Reply:
Terminal needs to separate the Application object from the Root Xaml Content, currently there one in the same which is problematic in the long run, especially if multiple threads with multiple Islands are involved.
For now this PR is not solving this issue.